### PR TITLE
feat(migration): add filter feedback during data export (#501)

### LINKED ISSUE #501: Add filter feedback during data export
## Problem

Users have no visibility into whether filters are being applied during data export. This can lead to unexpected behavior when filters are silently ignored (see #500).

## Real-World Impact

| With Filter | Without Filter |
|-------------|----------------|
| 269,315 records (2:43) | 4,962,562 records (7:09) |

An **18x difference** in exported records with zero feedback about filter status.

## Suggested Enhancement

Log when filters are applied:

```
Exporting customeraddress (filter: country='US' AND et_legacysiebelrowid IS NOT NULL)...
```

Or at minimum, show filter status in progress output:

```
Exporting customeraddress... (filtered)
Exporting customeraddress... (no filter)
```

## Bonus: Warn on Empty Filter

If an entity has a `<filter>` element in the schema but `FetchXmlFilter` is empty after parsing, warn the user that the filter may be malformed.

## Files

- `src/PPDS.Migration/Export/ParallelExporter.cs`

### PR DESCRIPTION
**Pull request by joshsmithxrm**
## Summary
- Add `FilterApplied` and `FilterDescription` properties to `ProgressEventArgs` so all progress consumers can see filter status per entity
- Show `(filtered)` suffix in export progress lines and log filter condition summary on first report per entity
- Warn when schema has an empty `<filter>` element (malformed filter that silently exports all records)

Closes #501

## Test Plan
- [x] `ProgressEventArgsTests.FilterApplied_DefaultsFalse` — defaults to false
- [x] `ProgressEventArgsTests.FilterApplied_CanBeSetAndRetrieved` — round-trip
- [x] `ConsoleProgressReporterTests.Report_ExportWithFilter_ShowsFilteredSuffix` — "(filtered)" appears
- [x] `ConsoleProgressReporterTests.Report_ExportWithFilter_ShowsFilterDescription` — condition text shown
- [x] `ConsoleProgressReporterTests.Report_ExportWithoutFilter_NoFilteredSuffix` — no "(filtered)" when unfiltered
- [x] `ConsoleProgressReporterTests.Report_ExportFilterDescription_ShownOnlyOnFirstReport` — no duplicate descriptions
- [x] `ParallelExporterFilterFeedbackTests` — 7 tests covering SummarizeFilter edge cases

## Verification
- [x] /gates passed
- [x] /review completed (findings: 2, both fixed)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/PPDS.Migration/Progress/ConsoleProgressReporter.cs [hunk 1]
```diff
@@ -80,13 +80,22 @@ public void Report(ProgressEventArgs args)
...
                         // Show success/failure breakdown if there are failures
                         var failureInfo = args.FailureCount > 0
                             ? $" [{args.SuccessCount} ok, {args.FailureCount} failed]"
                             : "";
 
-                        Console.Error.WriteLine($"{prefix} [{phase}] {args.Entity}{relInfo}{tierInfo}: {args.Current:N0}/{args.Total:N0}{pct}{rps}{eta}{failureInfo}");
+                        Console.Error.WriteLine($"{prefix} [{phase}] {args.Entity}{relInfo}{tierInfo}{filterInfo}: {args.Current:N0}/{args.Total:N0}{pct}{rps}{eta}{failureInfo}");
+
+                        // Show filter description on first progress report for this entity
+                        if (isNewEntity && args.FilterApplied && !string.IsNullOrEmpty(args.FilterDescription))
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Migration/Progress/ConsoleProgressReporter.cs:93]: Acknowledged. `ConsoleProgressReporter` is documented as not thread-safe (class docstring lines 10-15: *This class is not thread-safe. It is designed for single-threaded CLI usage where progress events are reported sequentially.*). The `_lastEntity` race condition is pre-existing and affects all progress tracking, not just the filter description. Worst case is cosmetic duplication of the filter line. A `ConcurrentDictionary` would fix just this case but the broader thread-safety gap remains — tracking as a separate concern.

File: src/PPDS.Migration/Export/ParallelExporter.cs [hunk 2]
```diff
@@ -257,6 +257,17 @@ private async Task<EntityExportResultWithData> ExportEntityAsync(
...
+            // Warn when schema has a <filter> element but it parsed to empty content
+            if (entitySchema.FetchXmlFilter != null && string.IsNullOrWhiteSpace(entitySchema.FetchXmlFilter))
+            {
+                progress.Report(new ProgressEventArgs
+                {
+                    Phase = MigrationPhase.Exporting,
+                    Entity = entitySchema.LogicalName,
+                    Message = $"Warning: {entitySchema.LogicalName} has a <filter> element in the schema but it is empty — all records will be exported (no filter applied). Check the schema for a malformed filter."
+                });
+            }
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Migration/Export/ParallelExporter.cs]: Good catch. Fixed in 3231c63 — now uses `SummarizeFilter` result to detect empty-but-structured filters (e.g., `<filter><filter></filter></filter>` with no conditions). The whitespace check remains as a fast-path for the simpler `<filter></filter>` case.

File: src/PPDS.Migration/Export/ParallelExporter.cs [hunk 3]
```diff
@@ -872,6 +893,42 @@ private string BuildFetchXml(EntitySchema entitySchema, int pageSize)
...
+                    var attr = c.Attribute("attribute")?.Value ?? "?";
+                    var op = c.Attribute("operator")?.Value ?? "?";
+                    var val = c.Attribute("value")?.Value;
+                    return val != null ? $"{attr} {op} '{val}'" : $"{attr} {op}";
+                }).ToList();
+
+                // Cap at 3 conditions to keep output concise
+                var summary = parts.Count <= 3
+                    ? string.Join(" AND ", parts)
+                    : string.Join(" AND ", parts.Take(3)) + $" (+{parts.Count - 3} more)";
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Migration/Export/ParallelExporter.cs]: Both points addressed in 3231c63:
1. Child `<value>` elements are now extracted for `in`/`not-in` operators — e.g., `statecode in '0,1'`
2. Changed separator from ` AND ` to `, ` to avoid misrepresenting OR filter logic as AND
